src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/GroovyMemoryLeakTest.java

```diff
@@ -35,18 +39,19 @@ public void clearLoaders() {
     private static final List<WeakReference<ClassLoader>> LOADERS = new ArrayList<>();
 
     public static void register(Object o) {
-        ClassLoader loader = o.getClass().getClassLoader();
-        System.err.println("registering " + o + " from " + loader);
-        LOADERS.add(new WeakReference<>(loader));
+        System.err.println("registering " + o);
+        for (ClassLoader loader = o.getClass().getClassLoader(); !(loader instanceof PluginManager.UberClassLoader); loader = loader.getParent()) {
+            System.err.println("…from " + loader);
+            LOADERS.add(new WeakReference<>(loader));
+        }
     }
 
     @Test
     public void loaderReleased() throws Exception {
-        Assume.assumeTrue(isRunningOnJDK8());
-
         FreeStyleProject p = r.jenkins.createProject(FreeStyleProject.class, "p");
+        String cp = CharacterPredicates.class.getProtectionDomain().getCodeSource().getLocation().toString(); // some JAR
         p.getPublishersList().add(new TestGroovyRecorder(
-                new SecureGroovyScript(GroovyMemoryLeakTest.class.getName()+".register(this)", false, null)));
+            new SecureGroovyScript(GroovyMemoryLeakTest.class.getName()+".register(this)", false, Collections.singletonList(new ClasspathEntry(cp)))));
```

Comment:
This assert should be verifying that the loader was released.  Okay.  

Should we still have a negative test for the the `PluginManager.UberClassLoader` where the release shouldn't/doesn't happen?

Reply:
It is not possible for the `Jenkins.pluginManager.uberClassLoader` to be released.